src/SFA.DAS.AssessorService.Web/Validators/Standard/OptOutStandardVersionViewModelValidator.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Linq;
+using FluentValidation;
+using FluentValidation.Results;
+using Microsoft.AspNetCore.Http;
+using SFA.DAS.AssessorService.Application.Api.Client.Clients;
+using SFA.DAS.AssessorService.Web.Helpers;
+using SFA.DAS.AssessorService.Web.ViewModels.Standard;
+
+namespace SFA.DAS.AssessorService.Web.Validators.Standard
+{
+    public class OptOutStandardVersionViewModelValidator : AbstractValidator<OptOutStandardVersionViewModel>
+    {
+        private string _standardVersion;
+    
+        public OptOutStandardVersionViewModelValidator(IHttpContextAccessor httpContextAccessor, IStandardVersionClient standardVersionApiClient)
+        {
+            var epaoOrgId = EpaOrgIdFinder.GetFromClaim(httpContextAccessor);
+
+            RuleFor(vm => vm.StandardReference)
```

Comment:
This should written to use the view model for both properties as it reads better and avoids potential although unlikely issues with concurrency using the instance variable to hold the version.

This works fine....

```
RuleFor(vm => vm.StandardReference)
                .NotEmpty()
                .Must((vm, _) =>
                {
                    
                    return standardVersionApiClient.GetEpaoRegisteredStandardVersions(epaoOrgId, vm.StandardReference)
                                                   .GetAwaiter()
                                                   .GetResult()
                                                   .FirstOrDefault(p => p.Version.Equals(vm.Version, StringComparison.InvariantCultureIgnoreCase)) != null;
                })
                .WithMessage(vm => $"You cannot opt out of {vm.StandardReference}, {vm.Version} as you have already opted out.");
```

Reply:
Done